cider/exec/template/CaseIR.cpp

```diff
@@ -158,11 +158,14 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCaseExpr(
   std::unique_ptr<CodegenColValues> case_val =
       codegenCaseExpr(case_expr, case_llvm_type, is_real_str, co);
   if (is_real_str) {
-    // FIXME(haiwei): [POAE7-2457] ,blocking by [POAE7-2415]
-    CIDER_THROW(CiderCompileException,
-                "String type case when is not currently supported.");
+    auto case_expr = dynamic_cast<FixedSizeColValues*>(case_val.get());
```

Comment:
should use `TwoValueColValues` type?

Reply:
and may need use new string type encoder and decoder.

Reply:
Could just use codegenVarCharColVar?

Reply:
Done.